src/main/java/net/snowflake/client/jdbc/DefaultResultStreamProvider.java

```diff
@@ -142,6 +144,23 @@ else if (context.getQrmk() != null) {
     return response;
   }
 
+  public static InputStream detectGzipAndGetStream(InputStream is) throws IOException {
+    try {
```

Comment:
we might even not need this, as the is comes wrapped in the correct decmpressor. this is handeled at the http client level.